### PR TITLE
Clean up project structure

### PR DESCRIPTION
**Pull request by jacob-hughes**
This one will only make sense when reviewing in chronological commit order :smile: 

The project was getting a bit unstructured with `lib.rs` growing and taking on code that it really shouldn't have. Now I have a pretty good idea about the API for `gcmalloc`, this reformat feels like it will make the project easier to navigate.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -19,6 +19,9 @@ extern crate packed_struct_codegen;
 pub mod alloc;
 pub mod gc;
 
+// Re-export public APIs
```
Comment (ltratt) [inline src/lib.rs]: I think the comment is redundant and can be removed.

**Comment by ltratt** [comment]
I have just that one minor comment.

File: src/lib.rs (hunk 1 above)
Comment (jacob-hughes) [reply src/lib.rs]: Fixed: 1da75a6e56d35720ebc243bc60d3a89bb63e7704

**Comment by ltratt** [comment]
Please squash.

**Comment by jacob-hughes** [comment]
Squashed

**Comment by ltratt** [comment]
I was assuming that https://github.com/softdevteam/gcmalloc/pull/25/commits/52645ef11a307bdf453272667ff6170ca4e9e11b would be the commit changed. Did I miss something?

**Comment by jacob-hughes** [comment]
Only me being an idiot and fixing up the commit into the latest one on auto pilot. Sorry, thanks for spotting.

**Comment by ltratt** [comment]
No problem -- force push an update when you're ready.

**Comment by jacob-hughes** [comment]
Squashed.

**Pull request merged**